include/openspace/interaction/delayedvariable.h

```diff
@@ -0,0 +1,84 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2017                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_CORE___DELAYEDVARIABLE___H__
+#define __OPENSPACE_CORE___DELAYEDVARIABLE___H__
+
+#include <ghoul/glm.h>
+
+namespace openspace {
+namespace interaction {
+
+/**
+ * Class that acts as a smoothing filter to a variable. The filter has a step
+ * response on a form that resembles the function y = 1-e^(-t/scale). The variable
+ * will be updates as soon as it is set to a value (calling the set() function).
+*/
+template <typename T, typename ScaleType>
+class DelayedVariable {
+public:
+    DelayedVariable(ScaleType scaleFactor, ScaleType friction) {
+        _scaleFactor = scaleFactor;
```

Comment:
Better to set these values in the initialization of the constructor, rather than afterwards.

Also use `std::move`.  It most cases, it's going to be an easy type, but it doesn't have to